kube/src/client/mod.rs

```diff
@@ -126,6 +153,19 @@ impl Client {
         &self.default_ns
     }
 
+    /// Perform a request described by a [`Verb`]
+    pub async fn call<V: Verb>(&self, verb: V) -> Result<<V::ResponseDecoder as TryFuture>::Ok, CallError>
+    where
+        <V::ResponseDecoder as TryFuture>::Error: std::error::Error + Send + Sync + 'static,
+    {
+        let req = verb.to_http_request().context(BuildRequestFailed)?;
+        V::ResponseDecoder::from(self.send(req).await.map_err(Box::new).context(RequestFailed)?)
+            .into_future()
+            .await
+            .map_err(|err| Box::new(err) as Box<dyn std::error::Error + Send + Sync + 'static>)
+            .context(DecodeFailed)
+    }
+
```

Comment:
This seems to be the main method here that's causing the abstraction to have to rewrite most of the `Client` methods.

Would it not make more sense to re-use the existing methods? Or is that going to be too difficult?
Not saying that this setup isn't nice or anything, but it's a big change, and you kind of need a full rewrite-the-world type change to get it in. It might be easier to start with `.call` just translating from `Verb` to existing `Client::method` for now.

Reply:
> Would it not make more sense to re-use the existing methods? Or is that going to be too difficult?

The existing methods aren't _too_ useful, and that would cause a circular dependency between `Client` and `Verb`. Not insurmountable, but it feels a bit smelly.

Reply:
Besides, everything except `request_status` and websockets is already implemented with the new `ResponseDecoder` mechanism, so it wouldn't really save much work to reuse them.